runtime/gc_structs/ClassIteratorClassSlots.cpp

```diff
@@ -46,29 +46,35 @@ GC_ClassIteratorClassSlots::nextSlot()
 
 	case classiteratorclassslots_state_constant_pool:
 		slotPtr = _constantPoolClassSlotIterator.nextSlot();
-		if(NULL != slotPtr) {
+		if (NULL != slotPtr) {
 			return slotPtr;
 		}
 		_state += 1;
 
 	case classiteratorclassslots_state_superclasses:
 		slotPtr = _classSuperclassesIterator.nextSlot();
-		if(NULL != slotPtr) {
+		if (NULL != slotPtr) {
 			return slotPtr;
 		}
 		_state += 1;
 
 	case classiteratorclassslots_state_interfaces:
 		slotPtr = _classLocalInterfaceIterator.nextSlot();
-		if(NULL != slotPtr) {
+		if (NULL != slotPtr) {
 			return slotPtr;
 		}
 		_state += 1;
 
 	case classiteratorclassslots_state_array_class_slots:
-		slotPtr = _classArrayClassSlotIterator.nextSlot();
-		if(NULL != slotPtr) {
-			return slotPtr;
+		/*
+		 * Checking sharedITable is an optimization that only checks booleanArrayClass Interfaces
+		 * since all array claseses share the same ITable.
+		 */
+		if (_shouldScanInterfaces) {
```

Comment:
I think an optimization is applied to wrong iterator. I believe it should be for one above, for `_classLocalInterfaceIterator`